### ISSUE TITLE
Random crashes when schematica is used

### ISSUE THREAD

**Issue by Faawkz** [Lunatrius/Schematica#375]
The game sometimes crashes when loading a schematic, sometimes when one is already loaded.
Help would be appreciated :)

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  CoreModBase (ShulkerBoxViewer-1.5-mc[1.11-1.12.2].jar)
  Inventory Tweaks Coremod (InventoryTweaks-1.63 (2).jar)
Contact their authors BEFORE contacting forge

// Surprise! Haha. Well, this is awkward.

Time: 8/18/18 9:22 AM
Description: Unexpected error

java.nio.BufferOverflowException
	at java.nio.Buffer.nextPutIndex(Buffer.java:521)
	at java.nio.DirectIntBufferU.put(DirectIntBufferU.java:297)
	at net.optifine.render.VboRegion.drawArrays(VboRegion.java:299)
	at net.minecraft.client.renderer.vertex.VertexBuffer.func_177358_a(VertexBuffer.java:50)
	at com.github.lunatrius.schematica.client.renderer.chunk.container.SchematicChunkRenderContainerVbo.func_178001_a(SchematicChunkRenderContainerVbo.java:28)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.renderBlockLayer(RenderSchematic.java:802)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.func_174977_a(RenderSchematic.java:793)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.renderWorld(RenderSchematic.java:352)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.renderSchematic(RenderSchematic.java:278)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.onRenderWorldLast(RenderSchematic.java:247)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_28_RenderSchematic_onRenderWorldLast_RenderWorldLastEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.client.ForgeHooksClient.dispatchRenderLast(ForgeHooksClient.java:168)
	at sun.reflect.GeneratedMethodAccessor69.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.optifine.reflect.Reflector.callVoid(Reflector.java:653)
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1946)
	at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1613)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1384)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1117)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at java.nio.Buffer.nextPutIndex(Buffer.java:521)
	at java.nio.DirectIntBufferU.put(DirectIntBufferU.java:297)
	at net.optifine.render.VboRegion.drawArrays(VboRegion.java:299)
	at net.minecraft.client.renderer.vertex.VertexBuffer.func_177358_a(VertexBuffer.java:50)
	at com.github.lunatrius.schematica.client.renderer.chunk.container.SchematicChunkRenderContainerVbo.func_178001_a(SchematicChunkRenderContainerVbo.java:28)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.renderBlockLayer(RenderSchematic.java:802)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.func_174977_a(RenderSchematic.java:793)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.renderWorld(RenderSchematic.java:352)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.renderSchematic(RenderSchematic.java:278)
	at com.github.lunatrius.schematica.client.renderer.RenderSchematic.onRenderWorldLast(RenderSchematic.java:247)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_28_RenderSchematic_onRenderWorldLast_RenderWorldLastEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.client.ForgeHooksClient.dispatchRenderLast(ForgeHooksClient.java:168)
	at sun.reflect.GeneratedMethodAccessor69.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.optifine.reflect.Reflector.callVoid(Reflector.java:653)
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1946)
	at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1613)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['Fawkz_'/245, l='MpServer', x=-264.52, y=30.47, z=4.10]]
	Chunk stats: MultiplayerChunkCache: 4157, 4157
	Level seed: 0
	Level generator: ID 01 - flat, ver 0. Features enabled: false
	Level generator options: 
	Level spawn location: World: (-56,82,-30), Chunk: (at 8,5,2 in -4,-2; contains blocks -64,0,-32 to -49,255,-17), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Level time: 7439 game time, 7439 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 1 total; [EntityPlayerSP['Fawkz_'/245, l='MpServer', x=-264.52, y=30.47, z=4.10]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:532)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2739)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:426)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 656432184 bytes (626 MB) / 1560281088 bytes (1488 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=16M
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	FML: MCP 9.42 Powered by Forge 14.23.4.2739 Optifine OptiFine_1.12.2_HD_U_E1 10 mods loaded, 10 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State     | ID               | Version                  | Source                                      | Signature                                |
	|:--------- |:---------------- |:------------------------ |:------------------------------------------- |:---------------------------------------- |
	| UCHIJAAAA | minecraft        | 1.12.2                   | minecraft.jar                               | None                                     |
	| UCHIJAAAA | mcp              | 9.42                     | minecraft.jar                               | None                                     |
	| UCHIJAAAA | FML              | 8.0.99.99                | forge-1.12.2-14.23.4.2739.jar               | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | forge            | 14.23.4.2739             | forge-1.12.2-14.23.4.2739.jar               | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | inventorytweaks  | 1.63+release.109.220f184 | InventoryTweaks-1.63 (2).jar                | 55d2cd4f5f0961410bf7b91ef6c6bf00a766dcbe |
	| UCHIJAAAA | journeymap       | 1.12.2-5.5.2             | journeymap-1.12.2-5.5.2.jar                 | None                                     |
	| UCHIJAAAA | lunatriuscore    | 1.2.0.42                 | LunatriusCore-1.12.2-1.2.0.42-universal.jar | None                                     |
	| UCHIJAAAA | schematica       | 1.8.0.169                | Schematica-1.12.2-1.8.0.169-universal.jar   | None                                     |
	| UCHIJAAAA | shulkerboxviewer | 1.5                      | ShulkerBoxViewer-1.5-mc[1.11-1.12.2].jar    | None                                     |
	| UCHIJAAAA | worldedit        | 6.1.8                    | worldedit-forge-mc1.12-6.1.8-dist.jar       | None                                     |

	Loaded coremods (and transformers): 
CoreModBase (ShulkerBoxViewer-1.5-mc[1.11-1.12.2].jar)
  com.dazo66.shulkerboxshower.asm.MainTransformer
Inventory Tweaks Coremod (InventoryTweaks-1.63 (2).jar)
  invtweaks.forge.asm.ContainerTransformer
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 388.13' Renderer: 'GeForce GTX 1050/PCIe/SSE2'
	Launched Version: 1.12.2-forge1.12.2-14.23.4.2739
	LWJGL: 2.9.4
	OpenGL: GeForce GTX 1050/PCIe/SSE2 GL version 4.6.0 NVIDIA 388.13, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: Faithful+1.12.2-rv4.zip, HDTEXT 1.1.zip (incompatible)
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i3-6100 CPU @ 3.70GHz
	OptiFine Version: OptiFine_1.12.2_HD_U_E1
	OptiFine Build: 20180628-111228
	Render Distance Chunks: 32
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.6.0 NVIDIA 388.13
	OpenGlRenderer: GeForce GTX 1050/PCIe/SSE2
	OpenGlVendor: NVIDIA Corporation
	CpuCount: 4
```

**Comment by Lunatrius** [comment]
Duplicate of #321, either switch the VBO setting to OFF or remove Optifine while using Schematica.
